### PR TITLE
Increase reduction at root when the best move does not change frequently

### PR DESCRIPTION
**Pull request by SFisGOD**
STC:
LLR: 2.94 (-2.94,2.94) {-0.25,1.25}
Total: 51320 W: 5159 L: 4956 D: 41205
Ptnml(0-2): 215, 3897, 17242, 4082, 224
https://tests.stockfishchess.org/tests/view/5faa072367cbf42301d6a767

LTC:
LLR: 2.98 (-2.94,2.94) {0.25,1.25}
Total: 15952 W: 762 L: 642 D: 14548
Ptnml(0-2): 8, 561, 6725, 667, 15
https://tests.stockfishchess.org/tests/view/5faa4c3567cbf42301d6a794

Bench: 3954692

### CODE REVIEW

**Comment by FauziAkram** [comment]
Congratulations!
Just a quick note, such a high drawing rate in LTC, is it luck, or is it something related to this patch?

**Comment by SFisGOD** [comment]
@FauziAkram Thanks. NNUE = true tests have high draw rate, see https://github.com/official-stockfish/Stockfish/issues/3148

File: src/search.cpp [hunk 1]
```diff
@@ -1161,14 +1161,18 @@ namespace {
...
+          // Increase reduction if other threads are searching this position
           if (th.marked())
               r++;
 
           // Decrease reduction if position is or has been on the PV (~10 Elo)
           if (ss->ttPv)
               r -= 2;
 
+          // Increase reduction at root and non-PV nodes when the best move does not change frequently
+          if (!(PvNode && !rootNode) && depth > 10 && thisThread->bestMoveChanges <= 2)
```
Comment (pb00068) [inline src/search.cpp:1173]: To me
`if ((rootNode || !PvNode) && depth > 10 && thisThread->bestMoveChanges <= 2)`
looks much more easy to read/understand.

Comment (pb00068) [reply src/search.cpp:1173]: it also corresponds 1:1 to what the comment says

**Comment by vondele** [comment]
@SFisGOD I lost the proper author in git in squashing the commit, and it slipped through the cracks.. sorry for that..

**Comment by SFisGOD** [comment]
@vondele @snicolet  Is it possible to amend the author of last commit? It is like my only source of dopamine these days. Sad to know that I can't see it listed here https://github.com/official-stockfish/Stockfish/commits?author=SFisGOD

**Comment by vondele** [comment]
Unfortunately not on master (i.e. no force pushes on master). I had the rebase/squash the branch on the previous commit as the pull request had the wrong base. In resolving the conflicts I lost the proper author... sorry for that.

**Comment by SFisGOD** [comment]
@vondele I hope it will not happen again. It's rare to have an Elo gainer to pass so should be more careful in my opinion.

**Comment by vondele** [comment]
yes, I'll try of course.

But in that case please help by having your PRs and tests against master only, not against other pending PRs, and condense them into a single commit. The alternative is that I ask everybody to rebase their PRs on master, and only accept those that are a single commit on top of latest master. I'm trying to make it easier for contributors by doing some manual work on their PRs. Sometimes mistakes happen.

**Comment by SFisGOD** [comment]
@vondele Yes, maybe you should have told me to do so and I would have gladly complied. Anyway, lesson learned so I will not hurry in making a PR anymore. I will just wait until all pending PRs are merged and then I'll open my pull request. That way, it will be easier for you.

**Comment by vondele** [comment]
yes, same here, but let's not worry too much about it. After all, you made the contribution, it progressed the project, which is what really counts.

**Comment by SFisGOD** [comment]
@snicolet Thank you so much snicolet!

**Comment by vondele** [comment]
unless this breaks abrok, fishtest and other workflows out there...

**Comment by vondele** [comment]
at least abrok doesn't seem to update to this commit...
There is a very good reason not to force push to the master repo of a public project honestly.

**Comment by vondele** [comment]
fishtest will presumably drain, as all workers reference master as that has now been erased.

**Comment by snicolet** [comment]
If I have broken anything, I'll take the blame, so at least both SFisGOD and yourself can have a third person to direct the bitterness to 🙂.

Maybe there is a chance that fishtest does not drain, however, as fishtest workers pull the sources to compile from individual user commits rather than from the official-stockfish master branch?

For instance, it seems that the commit in question was with a SHA beginning with `bbed6...` and these commits still exist on github, both on the official-stockfish repository and and the snicolet repository:
https://github.com/snicolet/Stockfish/commit/bbed64a5345351221b9e2982644e16b3e13b0087
https://github.com/official-stockfish/Stockfish/commit/bbed64a5345351221b9e2982644e16b3e13b0087

But if I have broken something, could I undo the force-push to restore the previous commit in the master branch?

**Comment by vondele** [comment]
We'll have to see... basically I don't know how this will impact all things. Abrok still seems stuck, fishtest might survive on the cached binaries, or by finding back somehow the commit that was erased. Reverting with another force push... I have no idea

However, I hope I'm not sounding too bitter, I'm just a little, because of the time lost for such a non-issue.

**Comment by MichaelB7** [comment]
great find, 2.5 Elo gain does not around come every day ...

**Pull request closed without merging**